calico-vpp-agent/cni/network_vpp.go

```diff
@@ -193,6 +193,18 @@ func (s *Server) AddVppInterface(podSpec *storage.LocalPodSpec, doHostSideConf b
 	// Select the first 11 characters of the containerID for the host veth.
 	tunTag := podSpec.NetnsName + "-" + podSpec.InterfaceName
 
+	for idx := range podSpec.HostPorts {
```

Comment:
Here you might exceed the length of the table, and also we want to expose the ports on all the container addresses. 
Maybe you could use a dedicated function returning the first address that matches 
````golang
for _, hostPort := range podSpec.HostPorts {
   for _, containerAddr := range podSpec.ContainerIps {
       if addrFamilyDiffers (containerAddr, hostPort.HostIP) == nil {
          continue
       }
       err = s.BindHostPort(hostPort, containerAddr)
       ...
   } 
}
````
For the error cleanup, we might want to have `s.BindHostPort` return the EntryID and do `hostPorts.EntryID = id` here. That way if one of the call fails, we can call delete on all the previously created entries.